app/scripts/lib/app-start.js

```diff
@@ -105,47 +101,57 @@ define(function (require, exports, module) {
 
     initializeConfig () {
       return this._configLoader.fetch()
-                    .then(_.bind(this.useConfig, this))
-                    .then(_.bind(this.initializeAble, this))
```

Comment:
This here was the problem. We started creating all these dependencies as soon as config was loaded, but there was no guarantee that l10n was ready. Now initializeDeps ensures config and l10n are created in parallel, then all subdependencies are generated afterwards.